targets/targets.go

```diff
@@ -322,10 +324,11 @@ func New(targetsDef *targetspb.TargetsDef, ldLister endpoint.Lister, globalOpts
 		globalLogger.Error("Unable to produce the base target lister")
 		return nil, fmt.Errorf("targets.New(): Error making baseTargets: %v", err)
 	}
-
+	targetDNSResolver, dnsIP := createOverrideDNSResolver(targetsDef)
+	t.DnsIP = dnsIP
 	switch targetsDef.Type.(type) {
 	case *targetspb.TargetsDef_HostNames:
-		st, err := staticTargets(targetsDef.GetHostNames())
+		st, err := staticTargets(targetsDef.GetHostNames(), targetDNSResolver)
 		if err != nil {
```

Comment:
I was thinking instead of recreating targetDNSResolver, can we just use t.resolver in this call and the other calls below?

```suggestion
		st, err := staticTargets(targetsDef.GetHostNames(), t.resolver)
		if err != nil {
```

Reply:
I think we would need to cast since t.resolver is of type endpoint.Resolver while the methods below want an object of type *dnsRes.Resolver.